org/mozilla/jss/crypto/Algorithm.java

```diff
@@ -112,18 +112,31 @@ public OBJECT_IDENTIFIER toOID() throws NoSuchAlgorithmException {
      * @return Returns <code>true</code> if the given Object can be used as a parameter.
      */
     public boolean isValidParameterObject(Object o) {
-        if( o == null ) {
+        if (o == null) {
+            // It is only valid to pass a null parameter if the length of the
+            // expected class list is zero. Otherwise, the usage is invalid.
             return (parameterClasses.length == 0);
         }
-        if( parameterClasses.length == 0 ){
+
+        if (parameterClasses.length == 0) {
+            // If we got an object parameter, but weren't expecting any, it
+            // is invalid.
             return false;
         }
```

Comment:
This case is actually covered by the subsequent loop. It's up to you if you want to remove it.